### PR TITLE
nix-serve: nix 2.0 fixes

### LINKED ISSUE #36321: nix-serve doesn't work with nix 2.0
## Issue description

I've been using nix-serve (via configuring `services.nix-serve` in `configuration.nix`) to allow my laptop to pull packages I've built on my desktop.

After updating the nixos-unstable channel this morning and getting the release of nix 2.0, I noticed that nix-serve was consuming a lot of my CPU.

By running `journalctl -u nix-serve.service --boot` I see many thousands of messages like:

```
Mar 05 12:56:58 alyamar nix-serve[4123]: Error while loading /nix/store/5407lxlsxs6ffb1y9qp0v7z3wplg36gd-nix-serve-0.2-7e09caa/libexec/nix-serve/nix-serve.psgi: Can't locate Nix/Config.pm in @INC (you may need to install the Nix::Config module) (@INC contains: <<<huge list of paths omitted>>>) at at /nix/store/5407lxlsxs6ffb1y9qp0v7z3wplg36gd-nix-serve-0.2-7e09caa/libexec/nix-serve/nix-serve.psgi line 2.
```

I think the CPU usage is nix-serve constantly being started by systemd and immediately crashing. (I was getting weirdly inconsistent CPU usage with both gnome-system-monitor and top agreeing that my system was under a fair bit of load overall, but only top showing some usage from nix-serve; presumably  that's because no individual process was living long enough to consume many resources as far as gnome was concerned, but many thousands of them were being lumped together more effectively by top?)

I believe the `Nix::*` perl modules were previously provided by the derivation for nix itself, but they've been removed from nix as of version 2.0. Seems like `nix-serve` needs to be rewritten for 2.0?

If that's not something that can be done immediately, we should probably do something about the broken service burning CPU in the background.

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.23, NixOS, 18.03pre130719.a66ce38acea (Impala)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.0`
 - channels(root): `"nixos-18.03pre130719.a66ce38acea"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

### PR DESCRIPTION
**Pull request by volth**
###### Motivation for this change

Fixes #36321

### CODE REVIEW

**Comment by Mic92** [comment]
@GrahamcOfBorg build nix-serve

File: pkgs/tools/package-management/nix-serve/default.nix [hunk 1]
```diff
@@ -18,7 +18,7 @@ stdenv.mkDerivation rec {
     inherit rev sha256;
   };
 
-  buildInputs = [ bzip2 perl nix ]
+  buildInputs = [ bzip2 perl nix nix.perl-bindings ]
```
Comment (ip1981) [inline pkgs/tools/package-management/nix-serve/default.nix]: AFAIK you do not need `bzip2 perl nix` here.

Although `perl` coudl be useful for basic sanity check: `perl -c "$src/nix-serve.psgi"`

Comment (volth) [reply pkgs/tools/package-management/nix-serve/default.nix]: It expects ```nix-store``` and ```bzip2``` on ```$PATH```

https://github.com/edolstra/nix-serve/blob/e4675e38ab54942e351c7686e40fabec822120b9/nix-serve.psgi#L54

Comment (volth) [reply pkgs/tools/package-management/nix-serve/default.nix]: Ah, that's covered in nixos/modules/services/networking/nix-serve.nix, so you are right could be removed from here

Comment (ip1981) [reply pkgs/tools/package-management/nix-serve/default.nix]: Runtime and build dependencies are different things.

And if you inject ${pkgs.foo}/bin/foo into output script, you do not need to add foo as a build dependency.

Comment (ip1981) [reply pkgs/tools/package-management/nix-serve/default.nix]: [I am sure](https://github.com/ip1981/nixsap/blob/master/modules/pkgs/nix-serve/default.nix) :)

Comment (volth) [reply pkgs/tools/package-management/nix-serve/default.nix]: Can you submit a PR?

Comment (ip1981) [reply pkgs/tools/package-management/nix-serve/default.nix]: It is not worthwhile on its own.

Although, an important thing is that `$PERL5LIB` is automatically defined by `buildInputs` _and_ expanded at build time.

Also that's a kind of bug that `${perlPackages.Starman}/bin/starman` can't find its own modules, thus `Starman` is also included.

**Pull request merged**